### PR TITLE
fix(telemetry): DSM/LemonRX sensor cleanup for Altimeter and Flight Pack Monitors

### LINKED ISSUE #4598: Lemon-RX G2/DSM Sensor Cleanup
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

The Lemon-RX G2 comes with Voltage and Current/Consumption attachments.  Spektrum groups sensor values in a single telemetry message based on what is needed to display on a single Telemetry screen.  For EdgeTX, we keep sensors individually. so that the user can pick and choose how to display it.    This different philosophy creates A LOT of sensors as well as duplicated in EdgeTX.

Examples:
1. "Alt" (Altitude, duplicated) gets duplicated because there is one message for plain Altitude, and another for Vario (Altitude + Vertical Speed).
2. Many Flight pack messages are for Dual Battery, so if only Bat#1 is used, we are creating a lot of empty Bat#2 data.
3. "RB1C" (Consumption, duplicated) can come in two different messages, PowerBox and Flight Pack Monitor.
4. "Curr" is a duplicate of  RB1A, but different name.. that is OK

image
image
image
image

Example of Two different screens in SpektrumTX:
Altitude message (Alt)  used for first screen.  Vario message (Alt + VSpeed) uses on second.
On EdgeTX, since are two independent messages containing "Alt", it creates duplicate "Alt".

IMG_4034
IMG_4033

### Describe the solution you'd like
Cleanup what we can without breaking compatibility with Genuine Spektrum Receivers (As well as OrangeRX).

Step 1: (Easy)
Most of the Bat#2 are easy to clean, as well as some that needs to be filtered by specific NULL/NO-DATA value.

Step 2: To remove duplicates,   Keep memory/context of the type of message received, and if one message is a superset of another, don't process the smaller message/value.  For example, the Altimeter Message only have "Alt", but the Vario message has "Alt"+"VSpd".  So in this case, the Altimeter will be ignored to avoid duplicates.

Working with FMak (Lemon-RX) to see what we can do on the RX side too. Would be nice that the RX don't report the data if the Sensor is not connected.. Right now is reporting Zeros, but could be NULL/NO-DATA.

### Describe alternatives you've considered

_No response_

### Additional context

Binding with "DSM-Auto" was not working, had to use "DSM-X2F".  The problem was not in EdgeTX, but in Multi-Module.
FIXED: https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/pull/918

### PR DESCRIPTION
**Pull request by frankiearzu**
Fixes #4598

## Summary of changes:
1. Hide "Flss" and "Hold" Sensors with populated with NO-DATA  (0x7FFF or 0xFFFF)
2. On PowerBox Telemetry (Voltage + Consumption), Only report data if the Sensor is connected (Volts > 0)
3. Handle Duplicate "Alt" sensor when RX reports separate messages for Altimeter (only)  and Vario  (Alt + VSpeed)
4. Handle Duplicate "RB1C" (Bat1 Consumption) if reported by Flight Pack Monitor Sensor and PowerBox in the same RX (Lemon)
5. Hide Bat2 data if Bat2 is not connected (PowerBox and FlightPack).  Most hardware only monitors 1 bat

**Note**:  On this scenario sometimes the duplicate sensors can show again:
1. Have Telemetry "Discover New" active, when we do "Reset Telemetry" on the main screen.
2. Have Telemetry "Discover New" active while binding. 

This is because we process the Altitude message before we know that we have a Vario message (Altitude + V Speed).  The duplicate sensor turns RED after a few seconds, and has to be deleted by the user.

## Testing:
Spektrum:   AR637T (altimeter) + Flight Pack Consumption sensor + Voltage Sensor
Lemon-RX G2 LM0087 (Altimeter) + Battery Sensors
Spectrum AR410 (only RX voltage)

### Regression Testing no external sensors
OrangeRX , AR636 , Blade 230s, AR631

NOTE: Screenshots taken before fixing the VSpeed units to respect radio configuration (English/Metric)

### LEMON RX: (no more duplicates)
image
image
image

### Spektrum AR637T:  (Gyro/As3X/SAFE sensors removed for clarity)
Note: Does not report RB1V (PowerBox message), only Flight Pack Consumption (RB1A,RB1C). A3 is flight pack voltage
image
image

### CODE REVIEW

File: radio/src/telemetry/spektrum.cpp [hunk 1]
```diff
@@ -724,6 +740,30 @@ void processSpektrumPacket(const uint8_t *packet)
...
+
+    else if (i2cAddress == I2C_PBOX) {
+      // hide mAh Consumption already reported in Fligh Pack message 
+      // No Bat Voltage, hide Voltage and Consumption
+      if (flightPackTelemetry && (sensor->startByte == 4 ||  sensor->startByte == 6)) {
+          continue;
+      else if ((sensor->startByte == 0 || sensor->startByte == 4) && // Bat1 Voltage
+          spektrumGetValue(packet + 4, 0, uint16) == 0) continue;
+      else if ((sensor->startByte == 2 || sensor->startByte == 6) && // Bat2 Voltage
+          spektrumGetValue(packet + 4, 2, uint16) == 0) continue;
```
Comment (gagarinlg) [inline radio/src/telemetry/spektrum.cpp:757]: Can youi please fix the indentation here and add curly brackets for each "if".
This is almost impossible to read if you do not already know what it does already.

**Review by gagarinlg — DISMISSED** [review]

**Comment by frankiearzu** [comment]
Just saw in my screenshots that has Altitude in Feet and Vertical Speed in m/s...   should be f/s.  (my radio is using English units)
Correcting other conversion inconsistencies.

Looking at other protocols, there is a lot of conversions missing.. probably this conversion code should be shared across all protocols and will reduce some code size too (will add an enhancement for it)

**Comment by pfeerick** [comment]
This _looks_ fine... I'll be testing momentarily  with AR637T and a LemonRX receiver (7ch stabiliser one IIRC). 

Is this 2.9 compatible as is? As given it is exclusive to Spektrum, it's low enough risk to include in 2.9.4.

**Comment by pfeerick** [comment]
LGTM with LM0086 (7Ch Telem Rx). Without XT60 energy meter went from 21 to 13 sensors. Plugging in the XT60 energy meter also kept it at 13, until battery was connected, at which point RB1V was populated. 

image

AR637T is unchanged, with 21 sensors both with and without this PR.

**Comment by frankiearzu** [comment]
@pfeerick: It works in 2.9.x and main/2.10 to.. so far that spectrum.cpp file is exactly the same. Have been using the 2.9 current branch self-build for flying.

**Comment by pfeerick** [comment]
Excellent, thanks for the confirm ;)

**Pull request merged**